### PR TITLE
DO NOT MERGE: feat: Technical review: @scope documentation

### LINKED ISSUE #17751: Content suggestion: @scope
## What is the new suggestion?
Document @scope, which is supported in Chrome 105.

## Why is it important or useful?

It is going to be supported and people are asking for it

How many pages are likely to be needed?

- [x] Create a @scope tutorial (new)
- [x] Update [Introducing the CSS Cascade](https://developer.mozilla.org/en-US/docs/Web/CSS/Cascade) to include @scope
- [x] Updated  [Cascade and inheritance](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Cascade_and_inheritance)
- [x] Update [cascade layers](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Cascade_layers) (just remove open and close comment in list)
- [ ] Update Live example to include scope in cascade and specificity tutorials: [CSS Cascade Skills test](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Cascade_tasks)
- [x] Update [specificity](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity) to include @scope
- [x] Update [`:scope`](https://developer.mozilla.org/en-US/docs/Web/CSS/:scope)

## Other supporting information

Test page: https://codepen.io/miriamsuzanne/pen/wvyBNXP works in chrome canary
https://drafts.csswg.org/css-cascade-6/#scoped-styles

### PR DESCRIPTION
**Pull request by chrisdavidmills**
Note: This technical review is now completed and approved. For the follow-on editorial review, see https://github.com/mdn/content/pull/30290

---

### Description

The `@scope` at-rule was shipped in Chrome 118 (see https://chromestatus.com/feature/5100672734199808). @ramiy very kindly created a first pass at the `@scope` reference page. This PR builds on that, adding further details and making some related changes to other existing pages.

Specifically, this PR:

- Updates the `@scope` page.
- Adds reference pages for `CSSScopeRule` and its `end` and `start` properties.
- Updates the `:scope` reference page to add information about `@scope`, and provide better examples.
- Updates all the CSS cascade-related pages on MDN to include appropriate mentions of `@scope`, as listed in https://github.com/mdn/content/issues/17751. I didn't update the skills test live example because it seemed a bit out of scope. There are many aspects of the cascade that are not tested on that page; it doesn't feel like adding a `@scope` example is a requirement of my work here. 

### Related issues and pull requests

Fixes https://github.com/mdn/content/issues/17751

### CODE REVIEW

File: files/en-us/web/api/cssscoperule/end/index.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+{{Specifications}}
+
+## Browser compatibility
+
+{{Compat}}
+
+## See also
+
+- [`@scope`](/en-US/docs/Web/CSS/@scope)
```
Comment (ramiy) [inline files/en-us/web/api/cssscoperule/end/index.md:33]:
```suggestion
- {{CSSxRef("@scope")}}
```

File: files/en-us/web/api/cssscoperule/index.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+{{Specifications}}
+
+## Browser compatibility
+
+{{Compat}}
+
+## See also
+
+- [`@scope`](/en-US/docs/Web/CSS/@scope)
```
Comment (ramiy) [inline files/en-us/web/api/cssscoperule/index.md:62]:
```suggestion
- {{CSSxRef("@scope")}}
```

File: files/en-us/web/api/cssscoperule/index.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+slug: Web/API/CSSScopeRule
+page-type: web-api-interface
+status:
+  - experimental
+browser-compat: api.CSSScopeRule
+---
+
+{{ APIRef("CSSOM") }}
+
+The **`CSSScopeRule`** interface of the [CSS Object Model](/en-US/docs/Web/API/CSS_Object_Model) represents a CSS [`@scope`](/en-US/docs/Web/CSS/@scope) at-rule.
```
Comment (ramiy) [inline files/en-us/web/api/cssscoperule/index.md:12]:
```suggestion
The **`CSSScopeRule`** interface of the [CSS Object Model](/en-US/docs/Web/API/CSS_Object_Model) represents a CSS {{CSSxRef("@scope")}} at-rule.
```

File: files/en-us/web/api/cssscoperule/start/index.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+{{Specifications}}
+
+## Browser compatibility
+
+{{Compat}}
+
+## See also
+
+- [`@scope`](/en-US/docs/Web/CSS/@scope)
```
Comment (ramiy) [inline files/en-us/web/api/cssscoperule/start/index.md:33]:
```suggestion
- {{CSSxRef("@scope")}}
```

File: files/en-us/web/css/cascade/index.md [hunk 5]
```diff
@@ -7,7 +7,7 @@ spec-urls: https://drafts.csswg.org/css-cascade/
 
 {{CSSRef}}
 
-The **cascade** is an algorithm that defines how user agents combine property values originating from different sources. The cascade defines the origin and layer that takes precedence when declarations in more than one [origin](#origin_types) or [cascade layer](/en-US/docs/Web/CSS/@layer) set a value for a property on an element.
+The **cascade** is an algorithm that defines how user agents combine property values originating from different sources. The cascade defines the origin and layer that takes precedence when declarations in more than one [origin](#origin_types), [cascade layer](/en-US/docs/Web/CSS/@layer), or [`@scope`](/en-US/docs/Web/CSS/@scope) block set a value for a property on an element.
```
Comment (ramiy) [inline files/en-us/web/css/cascade/index.md:10]:
```suggestion
The **cascade** is an algorithm that defines how user agents combine property values originating from different sources. The cascade defines the origin and layer that takes precedence when declarations in more than one [origin](#origin_types), [cascade layer](/en-US/docs/Web/CSS/@layer), or {{CSSxRef("@scope")}} block set a value for a property on an element.
```

File: files/en-us/web/css/specificity/index.md [hunk 6]
```diff
@@ -237,6 +237,33 @@ footer a {
 }
 ```
 
+### How @scope blocks affect specificity
+
+Including a ruleset inside a [`@scope`](/en-US/docs/Web/CSS/@scope) block does not directly affect the specificity of its selector, regardless of the selectors used inside the scope root and limit.
```
Comment (ramiy) [inline files/en-us/web/css/specificity/index.md:242]:
```suggestion
Including a ruleset inside a {{CSSxRef("@scope")}} block does not directly affect the specificity of its selector, regardless of the selectors used inside the scope root and limit.
```

File: files/en-us/learn/css/building_blocks/cascade_and_inheritance/index.md [hunk 7]
```diff
@@ -267,6 +267,23 @@ Let's discuss a few things from the above example to understand what's happening
 
 If you change the first line of CSS in this example to read `@layer secondLayer, firstLayer;`, you will change the layer declaration order, and all the important styles from `firstLayer` will be changed to their respective values in `secondLayer`.
 
+### Scoping proximity
+
+Another advanced topic that you might not use right away but may need to understand in the future is [`@scope`](/en-US/docs/Web/CSS/@scope). This is an [at-rule](/en-US/docs/Web/CSS/At-rule) that enables you to create a block of rules that will only apply to a specific subsection of the HTML on your page. For example, to specify some styles that will only apply to {{htmlelement("img")}} elements nested inside elements with a class of `feature`:
```
Comment (ramiy) [inline files/en-us/learn/css/building_blocks/cascade_and_inheritance/index.md:272]:
```suggestion
Another advanced topic that you might not use right away but may need to understand in the future is {{CSSxRef("@scope")}}. This is an [at-rule](/en-US/docs/Web/CSS/At-rule) that enables you to create a block of rules that will only apply to a specific subsection of the HTML on your page. For example, to specify some styles that will only apply to {{htmlelement("img")}} elements nested inside elements with a class of `feature`:
```

**Comment by bramus** [comment]
Looks good overall, except for these parts on specificity:

- https://pr30132.content.dev.mdn.mozit.cloud/en-US/docs/Web/CSS/@scope#specificity_in_scope
- https://pr30132.content.dev.mdn.mozit.cloud/en-US/docs/Web/CSS/Specificity#how_scope_blocks_affect_specificity

I’m talking specifically about this example:

```css
@scope (.article-body) {
  /* here, img has a specificity of 0-1-0 + 0-0-1 = 0-1-1,
     as it has :scope implicitly prepended to it
     In other words, this img is equivalent to :scope img */
  img { ... }
}
```

In that example, the specificity of `img` is `0-0-1`. It’s only when `:scope` is explicitly added that its specificity gets added to the overall specificity.

See https://developer.chrome.com/articles/at-scope/#scope-and-specificity for details.

**Comment by chrisdavidmills** [comment]
> Looks good overall, except for these parts on specificity:

@bramus oops, I totally misunderstood that bit! Thanks for the review; I've pushed a fix.

Let me know if there is anything else you think needs attention here.

File: files/en-us/web/css/@scope/index.md [hunk 8]
```diff
@@ -9,69 +9,453 @@ browser-compat: css.at-rules.scope
...
+    background: rebeccapurple;
+    color: antiquewhite;
+    font-family: sans-serif;
+  }
+}
+```
+
+`:scope` is implicitly prepended to all scoped style rules. If you wish to, you can write it explicitly, or you could prepend the [nesting](/en-US/docs/Web/CSS/CSS_nesting) selector (`&`) to get the same effect. (You might find these styles easier to understand.)
+
+As such, the three lines in the following block are all equivalent:
```
Comment (bramus) [inline files/en-us/web/css/@scope/index.md:131]:
```suggestion
As such, the three lines in the following block are all equivalent in what they select:
```

File: files/en-us/web/css/@scope/index.md [hunk 9]
```diff
@@ -9,69 +9,453 @@ browser-compat: css.at-rules.scope
...
+      border: 5px solid black;
+      background-color: goldenrod;
+    }
+  }
+  ```
+
+### Specificity in `@scope`
+
+Including a ruleset inside a `@scope` block does not affect the specificity of its selector, regardless of the selectors used inside the scope root and limit.
+
```
Comment (bramus) [inline files/en-us/web/css/@scope/index.md:175]: I would add an example here:

```suggestion

```css
@scope (.article-body) {
  /* img has a specificity of 0-0-1 */
  img { ... }
}
```

```

File: files/en-us/web/css/specificity/index.md [hunk 10]
```diff
@@ -237,6 +237,31 @@ footer a {
 }
 ```
 
+### How `@scope` blocks affect specificity
+
+Including a ruleset inside a `@scope` block does not affect the specificity of its selector, regardless of the selectors used inside the scope root and limit.
+
```
Comment (bramus) [inline files/en-us/web/css/specificity/index.md:243]: I would add an example here:

```suggestion

```css
@scope (.article-body) {
  /* img has a specificity of 0-0-1 */
  img { ... }
}
```

```

**Comment by bramus** [comment]
Added three comments to add some clarifications. I’ll let you decide if you want to include them or not. Otherwise LGTM.

File: files/en-us/web/css/@scope/index.md [hunk 11]
```diff
@@ -9,69 +9,453 @@ browser-compat: css.at-rules.scope
...
+   │  ├─ p
+   │  ├─ img
+   │  ├─ p
+   │  ├─ figure
+   │     ├─ img
+   │     ├─ figcaption
+   │
+   ├─ footer
+      ├─ p
+      ├─ img
```
Comment (ramiy) [inline files/en-us/web/css/@scope/index.md:62]: Minor change:

```suggestion
body
└─ article.feature
   ├─ section.article-hero
   │  ├─ h2
   │  └─ img
   │
   ├─ section.article-body
   │  ├─ h3
   │  ├─ p
   │  ├─ img
   │  ├─ p
   │  └─ figure
   │     ├─ img
   │     └─ figcaption
   │
   └─ footer
      ├─ p
      └─ img
```

Comment (chrisdavidmills) [reply files/en-us/web/css/@scope/index.md:62]: Much better, thank you @ramiy!

File: files/en-us/web/css/@scope/index.md (hunk 9 above)
Comment (chrisdavidmills) [reply files/en-us/web/css/@scope/index.md:175]: Good call; added in my next commit

File: files/en-us/web/css/specificity/index.md (hunk 10 above)
Comment (chrisdavidmills) [reply files/en-us/web/css/specificity/index.md:250]: added

**Comment by chrisdavidmills** [comment]
Thanks for the review comments, @bramus and @ramiy. I'm now considering this technical review complete and will close this PR. I'll then open up a new PR based on the same branch, which will contain the editorial review for the PR.
